### ISSUE TITLE
What is your operating environment and the version of the software?

### ISSUE THREAD

**Issue by Outhbreaker** [azadis/MC-GAN#7]
I want to reproduce your experiment. But there have been some problems in the debugging process. There are even code errors. I think this will definitely not be your problem. So I want to ask the specific environment running version (Ps. My environment is configured according to Readme)
Such as:  tensor_gt[batch,0,:,:] = tensor_all.data[batch,self.out_id[batch],:,:]  
IndexError: list index out of range

**Comment by azadis** [comment]
I think the issue is due to a wrong batch size specified here https://github.com/azadis/MC-GAN/blob/master/scripts/train_StackGAN.sh#L37
This batch size should be equal to the number of letters of your input $DATA. For instance, for a given word FROZEN with its own font style, BATCHSIZE should be set to 6.

**Comment by Outhbreaker** [comment]
I am so happy that I could receive your reply so soon.

Then I want to talk about my current understanding. It may be that I have a problem with the understanding of the project or what misunderstanding the configuration of the environment, which leads to various states.

Before I used the old version of the pytorch，the problem I will put the picture in the attachment.And also I  send the new problem to you.

This is my current environment configuration (Ubuntu 16.04 + torch 1.0 + torchvision 0.21 others using pip install)

After modifying the environment version, I probably changed 3 bugs to make train_cGAN.sh run successfully,so I think it should be caused by some deviations from my version. I want to know the specific environment configuration on your side.

Today, I follow your introduction to open  train_StackGAN.sh and modify the batchsize=5. And I execute the statement "./scripts/train_StackGAN.sh BLOOD", but the problem is still here.

After that，I also want to know README.md: ./scripts/train_StackGAN.sh ${DATA} this ${DATA} should be replaced by the specific folder name such as BLOOD If do that, that means lots of folders need run one by one. 
![new_error](https://user-images.githubusercontent.com/36978385/49910148-92807900-febc-11e8-878a-46cc9b360f6c.png)
![old_error](https://user-images.githubusercontent.com/36978385/49910149-93190f80-febc-11e8-957a-b8bf4cbd9b9e.png)

In short, I am very grateful to hear from you and I will look forward to your next reply.

**Comment by azadis** [comment]
Sorry for the confusion. For BLOOD, I think you should set your batch size to 4. This will be the number of unique given letters. 
I don't think there is an issue with your environment and torch or torchvision versions.

The answer to your second question is actually yes. You need to train the end-to-end model for transferring the style of each font separately. This will be done through the train_StackGAN.sh file with the name of your data folder. Prior to that, you should train the Glyph Network on all fonts for learning the glyph shapes by running train_cGAN.sh.

**Comment by Outhbreaker** [comment]
Thanks for your answer. I have finished running train_cGAN.sh. And I also take the photo of running test_cGAN.sh. I follow your suggestion to change the batch size, but the problem is still there. Sorry, I don't want to bother you. I really don't know how to solve it. By the way, such as ft2_2, is this folder batch size is the same as the BlOOD? I will look forward to your reply.
![test_result](https://user-images.githubusercontent.com/36978385/50049652-de355b80-0124-11e9-8cf8-4704267db92e.png)
![batch_size](https://user-images.githubusercontent.com/36978385/50049590-96fa9b00-0123-11e9-9318-0475a9b5940f.png)
![before running](https://user-images.githubusercontent.com/36978385/50049591-9eba3f80-0123-11e9-9a35-b12162f0576f.png)
![running_result](https://user-images.githubusercontent.com/36978385/50049592-9eba3f80-0123-11e9-9d22-bfa0e132db86.png)

**Comment by azadis** [comment]
I think your issue is with the running path or data path. You are probably either running the code from a wrong path (do you run ```bash ./scripts/train_StackGAN.sh BLOOD``` ?) or your in-the-wild fonts are not downloaded to the correct path. Have you run ```./datasets/download_font_dataset.sh public_web_fonts``` ? BLOOD directory should exist under ```../datasets/public_web_font```.

**Comment by Outhbreaker** [comment]
I downloaded the date except the optional Others  have already downloaded in my repertory.
I show you  the data picture.
![dataset](https://user-images.githubusercontent.com/36978385/50412199-c6568980-0840-11e9-9588-2504ade649d1.png)
![data_a test](https://user-images.githubusercontent.com/36978385/50412228-eab26600-0840-11e9-96c1-5c1926222dd2.png)
Also, I follow what you said (In fact as before I did it the same as what you said )but the result is still as usual.
![result_run](https://user-images.githubusercontent.com/36978385/50413050-25b79800-0847-11e9-9370-30d317534fac.png)
![result_error](https://user-images.githubusercontent.com/36978385/50413064-57306380-0847-11e9-8144-9d32f9afbede.png)

**Comment by liukboy** [comment]
@Outhbreaker I meet the same issue with you,and now how do you settled it?Thank you.

**Comment by Outhbreaker** [comment]
@liukboy  I don't know how to do it. I still wait the reply from the author. By the way, are you Chinese college students？

**Comment by azadis** [comment]
The ```datasets``` directory containing the font files should not be under MC-GAN! Your current font files path is not correct. You should be able to do the following:
```cd path_to_MC-GAN```
```ls ../datasets/public_web_fonts ```
